main/tests/server/src/com/google/refine/importers/SeparatorBasedImporterTests.java

```diff
@@ -632,6 +641,39 @@ public void readCsvWithPropertiesIgnoreQuotes() {
         Assert.assertEquals((String) project.rows.get(0).cells.get(3).value, "data4");
     }
 
+    // ---------------------guess separators------------------------
+
+    @Test
+    public void testThatDefaultGuessIsATabSeparator() {
+        ObjectNode options = SUT.createParserUIInitializationData(
+                job, new LinkedList<>(), "text/json");
+        assertEquals("\\t", options.get("separator").textValue());
+    }
+
+    @Test
+    public void testThatSeparatorIsGuessedCorrectlyForCSV() throws IOException {
+        List<ObjectNode> fileRecords = prepareFileRecords("food.small.csv");
+        ObjectNode options = SUT.createParserUIInitializationData(
+                job, fileRecords, "text/csv");
+        assertEquals(",", options.get("separator").textValue());
+    }
+
+    @Test
+    public void testThatSeparatorIsGuessedCorrectlyForTSV() throws IOException {
+        List<ObjectNode> fileRecords = prepareFileRecords("movies-condensed.tsv");
+        ObjectNode options = SUT.createParserUIInitializationData(
+                job, fileRecords, "text/tsv");
+        assertEquals("\\t", options.get("separator").textValue());
+    }
+
+    @Test
+    public void testThatSeparatorIsGuessedCorrectlyForJSON() throws IOException {
+        List<ObjectNode> fileRecords = prepareFileRecords("grid_small.json");
+        ObjectNode options = SUT.createParserUIInitializationData(
+                job, fileRecords, "text/json");
+        assertEquals("\\t", options.get("separator").textValue());
```

Comment:
I am not sure why the guesser should return `\t` when run on a JSON file. Because a JSON file will not be parseable as a CSV anyway, I would not assert anything about the separator guessed in this case.

Reply:
I see. `\t` is guessed as separator by default in either case. I can remove this test.